doc/nrf/gs_modifying.rst

```diff
@@ -181,9 +257,27 @@ Configuring build types
 
 .. build_types_overview_start
 
-Build types enable you to use different sets of configuration options for each board.
-You can create several build type :file:`.conf` files per board and select one of them when building the application.
-This means that you do not have to use one :file:`prj.conf` file for your project and modify it each time to fit your needs.
+The application is defined by a set of configuration files providing software and hardware relayed definitions.
+This can be abbreviated as ``configuration set`` of the application.
+The configuration set specifies which features application uses and how it behaves.
+It is convenient to specify multiple configuration sets to quickly switch application behavior.
+The most common example is to switch between ``release`` and ``debug`` variant of the application.
+
+The configuration set that will be used during the application construction is selected by providing a ``build type``.
+In the |NCS| you can define multiple build types.
+This enable you to use different sets of configuration options for each board.
+
+In the |NCS| the :file:`prj.conf` file is the main software related configuration source owned by the application.
+You can provide additional files with software configuration keeping the name pattern :file:`prj_myname.conf`.
+Each provided :file:`prj.conf` file is meant to be used independently (they don't overlay each other).
```

Comment:
Worth defining the relationship between the configuration files and configuration sets: "Each provided :file:`prj_mybuildtype.conf` file results in a distinct configuration set with build type ``mybuildtype``"

Reply:
I tried to do it but seems. There should be some description and an example later in the text. I wanted to introduce it gradually. It may be that this failed :(

Reply:
I mean, it can be inferred from the example, but maybe it could be put more clearly.